src/RedisAuthenticationInfo.php

```diff
@@ -56,7 +40,7 @@ public static function fromRedisOptions(RedisOptions $options): self|null
      *
      * @return array{0:non-empty-string,1?:non-empty-string}
      */
-    public function toRedisAuthInfo(): array
+    public function toArray(): array
```

Comment:
I do not mind this, so if that is okay for u, I do not want to discuss this too much.
Just my 2 cents on this kind of naming: I am a fan of having explicit namings. 

There are a bunch of methods called "toArray" in several classes. When you call `toArray` on a `Config` (lets take `RedisOptions` as an example), then you receive a map having properties => value pairs.

Calling `toArray` on a doctrine collection, you get some array with generic key and a value. In ORM context, *ToMany collections, its always some integer - thats why they introduced `getValues` I guess, since you are not able to have it inferred a list. So another behavior, its not returning properties as we can see.

So obviously, `toArray` does not expose - based on its method name - what it does and yet everyone uses it for whatever reason.

On the other hand, `toRedisAuthInfo` does actually expose what this is about. It does actually provide the auth info value (thats how redis [names it internally](https://github.com/phpredis/phpredis/blob/5156e0320242ff05f327a3801667140069688c0e/library.c#L4399)).
Sure, we do not provide the actual "type" of what is being returned, but we also do not have methods called `getUsername()->toString()`, so it seems acceptable to have methods not exposing their return type by method name, right? :D 

Just chit chatting here, so thats why I personally do not like this kind of naming but since that is personal pref, I do not care if you want to call it `toArray`.